backend/src/schema/resolvers/users.spec.js

```diff
@@ -450,12 +593,112 @@ describe('DeleteUser', () => {
           })
         })
 
-        describe('deletion of all comments requested', () => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-14T11:11:54Z" title="Friday, February 14th 2020, 12:11:54 pm +01:00">Feb 14, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -244,7 +244,435 @@ describe('UpdateUser', () => {
   })
 })
 
-describe('DeleteUser', () => {
+describe('DeleteUser as another user', () => {
+  const deleteUserMutation = gql`
+    mutation($id: ID!, $resource: [Deletable]) {
+      DeleteUser(id: $id, resource: $resource) {
+        id
+        name
+        about
+        deleted
+        contributions {
+          id
+          content
+          contentExcerpt
+          deleted
+          comments {
+            id
+            content
+            contentExcerpt
+            deleted
+          }
+        }
+        comments {
+          id
+          content
+          contentExcerpt
+          deleted
+        }
+      }
+    }
+  `
+  beforeEach(async () => {
+    variables = { id: ' u343', resource: [] }
+
+    user = await Factory.build('user', {
+      name: 'My name should be deleted',
+      about: 'along with my about',
+      id: 'u343',
+    })
+  })
+
+  beforeEach(async () => {
+    const anotherUser = await Factory.build(
+      'user',
+      {
+        role: 'user',
+      },
+      {
+        email: 'user@example.org',
+        password: '1234',
+      },
+    )
+
+    authenticatedUser = await anotherUser.toJson()
+  })
+
+  it('a user has no authorization to delete another user accounts', async () => {
+    const { errors } = await mutate({ mutation: deleteUserMutation, variables })
+    expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
+  })
+})
+
+describe('DeleteUser as moderator', () => {
+  const deleteUserMutation = gql`
+    mutation($id: ID!, $resource: [Deletable]) {
+      DeleteUser(id: $id, resource: $resource) {
+        id
+        name
+        about
+        deleted
+        contributions {
+          id
+          content
+          contentExcerpt
+          deleted
+          comments {
+            id
+            content
+            contentExcerpt
+            deleted
+          }
+        }
+        comments {
+          id
+          content
+          contentExcerpt
+          deleted
+        }
+      }
+    }
+  `
+  beforeEach(async () => {
+    variables = { id: ' u343', resource: [] }
+
+    user = await Factory.build('user', {
+      name: 'My name should be deleted',
+      about: 'along with my about',
+      id: 'u343',
+    })
+  })
+
+  beforeEach(async () => {
+    const moderator = await Factory.build(
+      'user',
+      {
+        role: 'moderator',
+      },
+      {
+        email: 'moderator@example.org',
+        password: '1234',
+      },
+    )
+
+    authenticatedUser = await moderator.toJson()
+  })
+
+  it('moderator is not allowed to delete other user accounts', async () => {
+    const { errors } = await mutate({ mutation: deleteUserMutation, variables })
+    expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
+  })
+})
+
+describe('DeleteUser as Admin', () => {
+  const deleteUserMutation = gql`
+    mutation($id: ID!, $resource: [Deletable]) {
+      DeleteUser(id: $id, resource: $resource) {
+        id
+        name
+        about
+        deleted
+        contributions {
+          id
+          content
+          contentExcerpt
+          deleted
+          comments {
+            id
+            content
+            contentExcerpt
+            deleted
+          }
+        }
+        comments {
+          id
+          content
+          contentExcerpt
+          deleted
+        }
+      }
+    }
+  `
+  beforeEach(async () => {
+    variables = { id: ' u343', resource: [] }
+
+    user = await Factory.build('user', {
+      name: 'My name should be deleted',
+      about: 'along with my about',
+      id: 'u343',
+    })
+  })
+
+  describe('authenticated as Admin', () => {
+    beforeEach(async () => {
+      const admin = await Factory.build(
+        'user',
+        {
+          role: 'admin',
+        },
+        {
+          email: 'admin@example.org',
+          password: '1234',
+        },
+      )
+      authenticatedUser = await admin.toJson()
+    })
+
+    describe('deleting a user account', () => {
+      beforeEach(() => {
+        variables = { ...variables, id: 'u343' }
+      })
+
+      describe('given posts and comments', () => {
+        beforeEach(async () => {
+          await Factory.build('category', {
+            id: 'cat9',
+            name: 'Democracy & Politics',
+            icon: 'university',
+          })
+          await Factory.build(
+            'post',
+            {
+              id: 'p139',
+              content: 'Post by user u343',
+            },
+            {
+              author: user,
+              categoryIds,
+            },
+          )
+          await Factory.build(
+            'comment',
+            {
+              id: 'c155',
+              content: 'Comment by user u343',
+            },
+            {
+              author: user,
+            },
+          )
+          await Factory.build(
+            'comment',
+            {
+              id: 'c156',
+              content: "A comment by someone else on user u343's post",
+            },
+            {
+              postId: 'p139',
+            },
+          )
+        })
+
+        it("deletes account, but doesn't delete posts or comments by default", async () => {
+          const expectedResponse = {
+            data: {
+              DeleteUser: {
+                id: 'u343',
+                name: 'UNAVAILABLE',
+                about: 'UNAVAILABLE',
+                deleted: true,
+                contributions: [
+                  {
+                    id: 'p139',
+                    content: 'Post by user u343',
+                    contentExcerpt: 'Post by user u343',
+                    deleted: false,
+                    comments: [
+                      {
+                        id: 'c156',
+                        content: "A comment by someone else on user u343's post",
+                        contentExcerpt: "A comment by someone else on user u343's post",
+                        deleted: false,
+                      },
+                    ],
+                  },
+                ],
+                comments: [
+                  {
+                    id: 'c155',
+                    content: 'Comment by user u343',
+                    contentExcerpt: 'Comment by user u343',
+                    deleted: false,
+                  },
+                ],
+              },
+            },
+            errors: undefined,
+          }
+          await expect(mutate({ mutation: deleteUserMutation, variables })).resolves.toMatchObject(
+            expectedResponse,
+          )
+        })
+
+        describe('deletion of all post requested', () => {
+          beforeEach(() => {
+            variables = { ...variables, resource: ['Post'] }
+          })
+
+          describe("marks user's posts as deleted", () => {
+            it('on request', async () => {
+              const expectedResponse = {
+                data: {
+                  DeleteUser: {
+                    id: 'u343',
+                    name: 'UNAVAILABLE',
+                    about: 'UNAVAILABLE',
+                    deleted: true,
+                    contributions: [
+                      {
+                        id: 'p139',
+                        content: 'UNAVAILABLE',
+                        contentExcerpt: 'UNAVAILABLE',
+                        deleted: true,
+                        comments: [
+                          {
+                            id: 'c156',
+                            content: 'UNAVAILABLE',
+                            contentExcerpt: 'UNAVAILABLE',
+                            deleted: true,
+                          },
+                        ],
+                      },
+                    ],
+                    comments: [
+                      {
+                        id: 'c155',
+                        content: 'Comment by user u343',
+                        contentExcerpt: 'Comment by user u343',
+                        deleted: false,
+                      },
+                    ],
+                  },
+                },
+                errors: undefined,
+              }
+              await expect(
+                mutate({ mutation: deleteUserMutation, variables }),
+              ).resolves.toMatchObject(expectedResponse)
+            })
+          })
+        })
+
+        describe('deletion of all comments requested', () => {
+          beforeEach(() => {
+            variables = { ...variables, resource: ['Comment'] }
+          })
+
+          it('marks comments as deleted', async () => {
+            const expectedResponse = {
+              data: {
+                DeleteUser: {
+                  id: 'u343',
+                  name: 'UNAVAILABLE',
+                  about: 'UNAVAILABLE',
+                  deleted: true,
+                  contributions: [
+                    {
+                      id: 'p139',
+                      content: 'Post by user u343',
+                      contentExcerpt: 'Post by user u343',
+                      deleted: false,
+                      comments: [
+                        {
+                          id: 'c156',
+                          content: "A comment by someone else on user u343's post",
+                          contentExcerpt: "A comment by someone else on user u343's post",
+                          deleted: false,
+                        },
+                      ],
+                    },
+                  ],
+                  comments: [
+                    {
+                      id: 'c155',
+                      content: 'UNAVAILABLE',
+                      contentExcerpt: 'UNAVAILABLE',
+                      deleted: true,
+                    },
+                  ],
+                },
+              },
+              errors: undefined,
+            }
+            await expect(
+              mutate({ mutation: deleteUserMutation, variables }),
+            ).resolves.toMatchObject(expectedResponse)
+          })
+        })
+
+        describe('deletion of all posts and comments requested', () => {
+          beforeEach(() => {
+            variables = { ...variables, resource: ['Post', 'Comment'] }
+          })
+
+          it('marks posts and comments as deleted', async () => {
+            const expectedResponse = {
+              data: {
+                DeleteUser: {
+                  id: 'u343',
+                  name: 'UNAVAILABLE',
+                  about: 'UNAVAILABLE',
+                  deleted: true,
+                  contributions: [
+                    {
+                      id: 'p139',
+                      content: 'UNAVAILABLE',
+                      contentExcerpt: 'UNAVAILABLE',
+                      deleted: true,
+                      comments: [
+                        {
+                          id: 'c156',
+                          content: 'UNAVAILABLE',
+                          contentExcerpt: 'UNAVAILABLE',
+                          deleted: true,
+                        },
+                      ],
+                    },
+                  ],
+                  comments: [
+                    {
+                      id: 'c155',
+                      content: 'UNAVAILABLE',
+                      contentExcerpt: 'UNAVAILABLE',
+                      deleted: true,
+                    },
+                  ],
+                },
+              },
+              errors: undefined,
+            }
+            await expect(
+              mutate({ mutation: deleteUserMutation, variables }),
+            ).resolves.toMatchObject(expectedResponse)
+          })
+        })
+      })
+
+      describe('connected `EmailAddress` nodes', () => {
+        it('will be removed completely', async () => {
+          await expect(neode.all('EmailAddress')).resolves.toHaveLength(2)
```


@ogerly similar to this test case it would be great if you could check with `neode` or a `neo4j` driver connection *directly in the database* if the data was really wiped out.

1. You take the `id` of the response or the `variables`
2. You run a database statement to check that the data is there
3. You run your delete mutation
4. You run the same database statement to check that the data *is not there* anymore

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-14T11:12:30Z" title="Friday, February 14th 2020, 12:12:30 pm +01:00">Feb 14, 2020</time>_

---


And I'm talking about the scope `deletion of all posts and comments requested` for example. :point_up: